payment/vote.go

```diff
@@ -210,35 +211,44 @@ func (service *Service) Vote(
 	// generate all the cb credential redemptions
 	requestCredentials, err := generateCredentialRedemptions(
 		context.WithValue(ctx, appctx.DatastoreCTXKey, service.datastore), credentials)
-	if err != nil {
-		return fmt.Errorf("error generating credential redemptions: %w", err)
-	}
 
-	// get a new VoteEvent to emit to kafka based on our input vote
-	voteEvent, err := NewVoteEvent(vote)
-	if err != nil {
-		return fmt.Errorf("failed to convert vote to kafka vote event: %w", err)
-	}
+	// foreach request credential, make vote
+	for _, rc := range requestCredentials {
+		// one vote per request credential
+		vote.VoteTally = 1
+		// set the funding source to the issuer
+		vote.FundingSource = rc.Issuer
 
-	// encode the event for processing
-	voteEventBinary, err := voteEvent.CodecEncode(service.codecs["vote"])
-	if err != nil {
-		return fmt.Errorf("failed to encode avro codec: %w", err)
-	}
+		if err != nil {
```

Comment:
Should we reset err here? If you insert something into the db and there is an err then when we enter the loop to access the next request credential this line will fail